app/controllers/students_controller.rb

```diff
@@ -46,20 +47,47 @@ def authorize_user
     authorize User
   end
 
+  def authorize_update
+    authorize @student 
+  end
+
   def student_params
-    params.require(:student).permit(
-      :cvu,
-      :name,
-      :paternal_last_name,
-      :maternal_last_name,
-      :rfc,
-      :curp,
-      :gender,
-      :marital_status,
-      :birth_date,
-      :country_birth,
-      :state_birth,
-    )
+    if current_user.is_admin_or_super_admin?
```

Comment:
There is plenty of duplicated code here. Please consider doing something as specified here: https://stackoverflow.com/questions/26306693/strong-parameters-how-to-permit-parameters-using-conditions

Creating a list of the permitted params by the user and another permitted by the admin and only append the lists if the user is admin.

Reply:
Done